pkg/util/clientauthorizer/admin_test.go

```diff
@@ -0,0 +1,161 @@
+package clientauthorizer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	utiltls "github.com/Azure/ARO-RP/pkg/util/tls"
+)
+
+func TestAdminClientAuthorizer(t *testing.T) {
+	caBundlePath := "/fake/path/to/ca/cert.pem"
+	log := logrus.NewEntry(logrus.StandardLogger())
+
+	validCaKey, validCaCerts, err := utiltls.GenerateKeyAndCertificate("validca", nil, nil, true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	validIntermediateCaKey, validIntermediateCaCerts, err := utiltls.GenerateKeyAndCertificate("valid-intermediate-ca", validCaKey, validCaCerts[0], true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	invalidCaKey, invalidCaCerts, err := utiltls.GenerateKeyAndCertificate("invalidca", nil, nil, true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, validSingleClientCert, err := utiltls.GenerateKeyAndCertificate("validclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, validCertWithIntermediates, err := utiltls.GenerateKeyAndCertificate("validclient", validIntermediateCaKey, validIntermediateCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, invalidCommonNameClientCert, err := utiltls.GenerateKeyAndCertificate("invalidclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, invalidKeyUsagesCert, err := utiltls.GenerateKeyAndCertificate("validclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	invalidKeyUsagesCert[0].ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth}
+
+	_, invalidExpiredCert, err := utiltls.GenerateKeyAndCertificate("validclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	invalidExpiredCert[0].NotAfter = time.Now().Add(-time.Hour)
+
+	_, invalidSigningCa, err := utiltls.GenerateKeyAndCertificate("validclient", invalidCaKey, invalidCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tt := range []struct {
+		name           string
+		wantCommonName string
```

Comment:
remove `want`.  Better, can move this into the body of the test as it's constant over all test cases.